spec/controllers/mount_controller_spec.rb

```diff
@@ -13,7 +13,7 @@
 
   describe '#alternate_url' do
     it 'returns an alternate_url if one is available' do
-      expect(subject).to receive(:url_for).with('script_name' => '/cy/alternative_engine')
+      expect(subject).to receive(:url_for).with('script_name' => '/cy/alternative_engine').and_return('/cy/alternative_engine')
```

Comment:
Would following do the trick?

`expect(subject).to receive(:url_for).with('script_name' => '/cy/alternative_engine').and_call_original`